src/Authorization.AspNetCore/AuthorizationErrorMessageBuilder.cs

```diff
@@ -0,0 +1,107 @@
+#nullable enable
+
+using System.Linq;
+using System.Text;
+using GraphQL.Language.AST;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Infrastructure;
+
+namespace GraphQL.Server.Authorization.AspNetCore
+{
+    public class AuthorizationErrorMessageBuilder : IAuthorizationErrorMessageBuilder
+    {
+        public string GenerateMessage(OperationType? operationType, AuthorizationFailure failure)
+        {
+            var error = new StringBuilder();
+            AppendFailureHeader(error, operationType);
+
+            foreach (var requirement in failure.FailedRequirements)
+            {
+                AppendFailureLine(error, requirement);
+            }
+
+            return error.ToString();
+        }
+
+        protected virtual string GetOperationType(OperationType? operationType)
```

Comment:
virtual?

Reply:
yes. maybe someone wants to override this with other text or other language.

Reply:
Then overriding `AppendFailureHeader` is enough.